test/unit/plugins/rest_spec.js

```diff
@@ -121,24 +170,114 @@ describe("The Rest plugin", function () {
 			});
 		});
 
-		describe("with invalid parameter(s) passed", function () {
-			it("returns an error with statusCode 400", function () {
+		describe("Ssh polling encounters an error", function () {
+			before(function () {
+				startSshPollingStub = Sinon.stub(sshAdapter, "startSshPolling")
+				.rejects(new Error("Simulated Failure"));
+
+				updateInstanceStub = Sinon.stub(instanceAdapter,"updateInstance")
+				.returns(Bluebird.resolve());
+
+				getPublicIPAddressStub = Sinon.stub(awsAdapter, "getPublicIPAddress", function () {
+					return (Bluebird.resolve(VALID_IP_ADDRESS));
+				});
+			});
+
+			after(function () {
+				getPublicIPAddressStub.restore();
+				startSshPollingStub.restore();
+				updateInstanceStub.restore();
+			});
+
+			it("fails", function () {
 				var request = new Request("POST", "/v1/instances").mime("application/json").payload({
-					ami  : INVALID_AMI,
+					ami  : VALID_AMI,
 					type : VALID_TYPE
 				});
 				return request.inject(server)
 				.then(function (response) {
-					var error = JSON.parse(response.payload);
-					expect(error.statusCode, "status").to.equal(400);
-					expect(error.message)
-					.to.equal("Bad Request: Please check the parameters passed.");
+					response.payload = JSON.parse(response.payload);
+					expect(response.statusCode, "status").to.equal(201);
+					expect(response.headers.location, "location").to.match(location);
```

Comment:
This test case doesn't look any different than the success case. What exactly do we expect to happen here that's different?